source/common/http/conn_manager_utility.cc

```diff
@@ -206,11 +206,13 @@ Network::Address::InstanceConstSharedPtr ConnectionManagerUtility::mutateRequest
   }
 
   // Generate x-request-id for all edge requests, or if there is none.
-  if (config.generateRequestId() && (edge_request || !request_headers.RequestId())) {
+  if (config.generateRequestId()) {
     // TODO(PiotrSikora) PERF: Write UUID directly to the header map.
-    const std::string uuid = random.uuid();
-    ASSERT(!uuid.empty());
-    request_headers.insertRequestId().value(uuid);
+    if ((!config.edgeAcceptRequestId() && edge_request) || !request_headers.RequestId()) {
```

Comment:
I feel like there are 8 cases here to cover in tests, as there are 8 combinations of these 3 bools we are combining.

It wasn't obvious to me that your tests covered all the cases.

Reply:
done